### PR TITLE
[GLUTEN-4058][CORE] Support JDK 11 and 17 for Gluten

### PR DESCRIPTION
**Pull request by surnaik**
## What changes were proposed in this pull request?

Support for JDK 11 and 17

Fixes: #4058 

## How was this patch tested?

Existing UTs

### CODE REVIEW

**Comment by surnaik** [comment]
@zhouyuan Please review. Thanks

**Comment by FelixYBW** [comment]
Thank you for your PR. Did you test with spark3.2, 3.3 and 3.4?

**Comment by surnaik** [comment]
> Thank you for your PR. Did you test with spark3.2, 3.3 and 3.4?

Thanks @FelixYBW for taking a look, yes, I have tested with 3.2 and 3.3. I haven't tested 3.4

**Comment by FelixYBW** [comment]
Is there performance difference?

**Comment by surnaik** [comment]
> Is there performance difference?

No, there isn't any significant difference

**Comment by FelixYBW** [comment]
Thank you for sharing. @zhouyuan @weiting-chen Can you have a review what should we do next?

**Comment by surnaik** [comment]
Test failure seems unrelated

**Comment by weiting-chen** [comment]
> Thank you for sharing. @zhouyuan @weiting-chen Can you have a review what should we do next?

Currently our CI system only run test with JDK8 and most of Gluten users are still using JDK8 as well.
We may consider to support JDK11 in next release.
If we decide moving to JDK11 and JDK17,we should also add CI system testing for JDK11 and 17 as well.
For JDK11, we can pick up this PR to run more testing and check if there is any issues for JDK11.
For JDK17, there was an unmerged PR(#2011 ) may also need to be added with some parameters, we should create a profile in pom file to manage which version to be used in Gluten.

**Comment by surnaik** [comment]
I see, thanks @weiting-chen. Sure, we can take this up later. Yes, parameter needs to be added, I thought of adding it directly to the CI run using -DargLine when the CI supports JDK 11/17

**Comment by zhouyuan** [comment]
@zzcclp @lgbo-ustc This patch tries to add JDK11 support, could you please kindly help to check the issue on CK backend?  There's a minor change on the shutdown hook
The log from CK backend showed a core dump on exiting:
```
13:16:23  # A fatal error has been detected by the Java Runtime Environment:
13:16:23  #
13:16:23  #  SIGSEGV (0xb) at pc=0x00007f302735b95e, pid=1020, tid=0x00007f2ff177f700
13:16:23  #
13:16:23  # JRE version: OpenJDK Runtime Environment (8.0_312-b07) (build 1.8.0_312-b07)
13:16:23  # Java VM: OpenJDK 64-Bit Server VM (25.312-b07 mixed mode linux-amd64 compressed oops)
13:16:23  # Problematic frame:
13:16:23  # C  [libch.so+0xf6b595e]  DB::CHJIT::deleteCompiledModule(DB::CHJIT::CompiledModule const&)+0xde
13:16:23  #
13:16:23  # Core dump written. Default location: /home/jenkins/agent/workspace/gluten/gluten-ci/gluten2/backends-clickhouse/core or core.1020
```

File: gluten-core/src/main/scala/org/apache/spark/util/GlutenShutdownManager.scala [hunk 1]
```diff
@@ -18,12 +18,18 @@ package org.apache.spark.util
 
 object GlutenShutdownManager {
 
+  private def isJava8: Boolean = System.getProperty("java.version").equals("8")
```
Comment (baibaichen) [inline gluten-core/src/main/scala/org/apache/spark/util/GlutenShutdownManager.scala:21]: @lwz9103 colud you please check whether we set this environment variable our CI environment?

Comment (surnaik) [reply gluten-core/src/main/scala/org/apache/spark/util/GlutenShutdownManager.scala:21]: This is a property exposed by the JVM, it should be present in all the JVM

Comment (surnaik) [reply gluten-core/src/main/scala/org/apache/spark/util/GlutenShutdownManager.scala:21]: But I think before Java 9, the java.version is set to 1.major.minor_subminor. I'll change this to parse the major version.

**Pull request closed without merging**